### PR TITLE
Fortran standard to 2008

### PR DESCRIPTION
**Pull request by gang525**
## Purpose
The Fortran standard is 2008 because petsc wasn't cooperating with earlier standards.

## Type of change
What types of change is it?
_Select the appropriate type(s) that describe this PR_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [x] Maintenance update
- [ ] Other (please describe)

## Testing
I do not have complex IDWarp compiled, so those tests were skipped.

## Checklist
_Put an `x` in the boxes that apply._

- [ ] I have run `flake8` and `black` to make sure the code adheres to PEP-8 and is consistently formatted
- [x] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

**Review by marcomangano — CHANGES_REQUESTED** [review]
Can you confirm these changes are intentional?

File: src/ADFirstAidKit/adBuffer.f [hunk 1]
```diff
@@ -841,9 +841,9 @@ SUBROUTINE POPREAL16(x)
       ENDIF
       END
 
-c======================= COMPLEX*8 =========================
+c======================= complex(kind=8) =========================
       BLOCK DATA COMPLEXS8
-      COMPLEX*8 adc8buf(512), adc8lbuf(512)
+      complex(kind=8) adc8buf(512), adc8lbuf(512)
```
[the remark this answers is not in the document]
Comment (gang525) [reply src/ADFirstAidKit/adBuffer.f:846]: I made a mistake with these. These should be complex(kind = 4). Thank you for catching this!

File: src/ADFirstAidKit/adBuffer.f [hunk 2]
```diff
@@ -934,9 +934,9 @@ SUBROUTINE POPCOMPLEX8(x)
       ENDIF
       END
 
-c======================= COMPLEX*16 =========================
+c======================= complex(kind=16) =========================
       BLOCK DATA COMPLEXS16
-      COMPLEX*16 adc16buf(512), adc16lbuf(512)
+      complex(kind=16) adc16buf(512), adc16lbuf(512)
```
[the remark this answers is not in the document]
Comment (gang525) [reply src/ADFirstAidKit/adBuffer.f]: Yea, I think I made the ctrl+f error during this entire file so I corrected everything with another commit shown below.

**Review by marcomangano — APPROVED** [review]

**Comment by nwu63** [comment]
I'm not 100% convinced we want to manually update the ADFirstAidKit stuff. A couple options here:
- do what's currently done in the PR, and then submit a patch upstream
- fix this here, but then refactor and move into a separate package (since multiple repos use the same code)
- modify the rules so we don't apply new Fortran standard here

Let's talk about this during the next maintenance meeting maybe.

**Comment by nwu63** [comment]
How does this look @eirikurj @bernardopacini? I added a new flag for F77 code which omits the standard specification. The flags can probably be simplified (we probably don't need PETSc/CGNS since it's just the Tapenade code).

I also see that @gang525 updated the `complexify` module. I assume this is unavoidable since we do not have a centralized complexify package (yet) and this is OK here. We will of course have to do the same thing for other repos that use the same module (but potentially have slightly different code).

**Review by eirikurj — APPROVED** [review]
Overall, this looks good. I got some feedback from the Tapenade people and according to them, future version of Tapenade will not include the `.f` files, only the C files, so eventually we could remove the F77 rules. For the time being, we can just keep this as is.

**Review by marcomangano — APPROVED** [review]
Up for the tapenade checks addition

**Review by nwu63 — APPROVED** [review]

**Pull request merged**